src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4942,10 +4942,10 @@ public ReadOnlyCollection<string> PayloadNames
             get
             {
                 // For contract based events we create the list lazily.
-                if (m_payloadNames == null)
+                // You can have m_payloadNames be null in the TraceLogging case (EventID < 0) so only
+                // do the lazy init if you know it is contract based (EventID >= 0)
+                if (EventId >= 0 && m_payloadNames == null)
```

Comment:
If there's precedent for returning null, it's certainly the cheaper option, so I'm ok with that.